lib/jekyll/utils/rouge.rb

```diff
@@ -4,16 +4,16 @@ module Rouge
 
       def self.html_formatter(*args)
         Jekyll::External.require_with_graceful_fail("rouge")
-        html_formatter = ::Rouge::Formatters::HTML.new(*args)
-        return html_formatter if old_api?
-
-        ::Rouge::Formatters::HTMLPygments.new(html_formatter)
+        if old_api?
+          ::Rouge::Formatters::HTML.new(*args)
+        else
+          ::Rouge::Formatters::HTMLLegacy.new(*args)
```

Comment:
Oh, sorry, this was unclear: HTMLLegacy also has line numbers and all features we currently use.
The 2.x version of the standard `HTML` formatter doesn't have line numbers.